src/ray/core_worker/core_worker.h

```diff
@@ -558,8 +557,11 @@ class CoreWorker {
   // Interface to submit non-actor tasks directly to leased workers.
   std::unique_ptr<CoreWorkerDirectTaskSubmitter> direct_task_submitter_;
 
+  mutable absl::Mutex actor_handles_mutex_;
```

Comment:
Please add a comment explaining why this is needed and what race conditions it protects against.

Reply:
Done.